### PR TITLE
feat(Tree): Added "padding" option to detail prop

### PR DESCRIPTION
**Pull request by ElliotTheHuman**
The `disablePadding` option allows you to disable the innate left and right padding on `ListItem`'s detail container.

Screen Shot 2021-05-18 at 4 39 13 PM

## Developer Checklist [ℹ️](https://github.com/looker-open-source/components/blob/main//CONTRIBUTING.md#developer-checklist)

- [ ] ♿️ Accessibility addressed
- [ ] 🌏 Localization & Internationalization addressed
- [ ] 🖼 Image Snapshot coverage
- [ ] 📚 Documentation updated
- [ ] 🧳 Bundle size impact addressed
- [ ] 🏁 Performance impacts addressed
- [ ] 👾 Browsers tested
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11

### CODE REVIEW

File: packages/components/src/Tree/Tree.tsx [hunk 1]
```diff
@@ -193,6 +193,7 @@ const TreeLayout: FC<TreeProps> = ({
       <TreeStyle
         border={hasBorder}
         branchFontWeight={branchFontWeight}
+        className={className}
```
Comment (ElliotTheHuman) [inline packages/components/src/Tree/Tree.tsx:196]: This line was accidentally removed during the keyboard nav PR merge. Without it, external style overrides can't be applied to `Tree` (of note, the border radius overrides in various stories).

File: packages/components/src/List/types.ts [hunk 2]
```diff
@@ -163,6 +163,10 @@ interface DetailOptions {
    * @default false
    */
   accessory?: boolean
+  /**
+   * If true, the innate left and and right padding on the detial container will be disabled
+   */
+  disablePadding?: boolean
```
Comment (lukelooker) [inline packages/components/src/List/types.ts:169]: I'd prefer this to be:

```suggestion
  /**
   * Padding is added to the left and and right of the detail container by default. You can
   * optionally disable that behavior if you want to have fine-grained control.
   * @default true
   */
  padding?: boolean
```

File: packages/components/src/Tree/Tree.tsx (hunk 1 above)
Comment (lukelooker) [reply packages/components/src/Tree/Tree.tsx:196]: eek!

**Review by lukelooker — DISMISSED** [review]
LGTM! :)

**Review by lukelooker — APPROVED** [review]
LGTM!

**Pull request merged**